### ISSUE TITLE
unrecognized error in addSecureConnection

### ISSUE THREAD

**Issue by 1dal** [phergie/phergie-irc-client-react#66]
Hi guys, 
i think [here](https://github.com/phergie/phergie-irc-client-react/blob/master/src/Client.php#L681) can be added some error handling/logging. Something like ```function ($e) { $this->logger->error($e->getMessage()); }``` can be very useful. 

Currently i was not able detect problem with SecureConnector (in my case `"Unable to complete SSL/TLS handshake: stream_socket_enable_crypto(): SSL operation failed with code 1. OpenSSL Error messages: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed"`. 

Thank your for your time, nice project!

**Comment by elazar** [comment]
@1dal Can you give us more information on the error or exception you experienced, like a backtrace?

[This line](https://github.com/phergie/phergie-irc-client-react/blob/master/src/Client.php#L679) is above the one you linked to:

```php
$this->initializeStream($stream, $connection);
```

`initializeStream()` makes [this call](https://github.com/phergie/phergie-irc-client-react/blob/master/src/Client.php#L697) to `configureStreams()`:

```php
$this->configureStreams($connection, $stream, $write);
```

And `configureStreams()` performs [these calls](https://github.com/phergie/phergie-irc-client-react/blob/master/src/Client.php#L577-L580), which should be setting callbacks for when the connection is terminated or encounters an error:

```php
$stream->on('end', $this->getEndCallback($read, $write, $connection, $timer));
$error = $this->getErrorCallback($connection);
$read->on('error', $error);
$write->on('error', $error);
```

If this isn't happening, then I agree that a change needs to be made so it does. However, we need more information before we can determine the issue and how to go about addressing it.

**Comment by 1dal** [comment]
@elazar thx for quick reply. Sorry, i was drunk when targeted linked line.

I meant something like this: https://github.com/phergie/phergie-irc-client-react/commit/28a877887b6efe7c3cdc7a772b1c5e9d5d1be368

I tried make secure connection over SOCKS with SSL, https://gist.github.com/1dal/3928412f9073e3f5bdfe0718bcc0b47d

...probably wrong, but it works for me. :) 

Thank you

**Comment by elazar** [comment]
@1dal 

> I meant something like this: 28a8778

This seems reasonable. Could you submit a PR for it?

> I tried make secure connection over SOCKS with SSL, ...probably wrong, but it works for me. :)

Looks reasonable to me. If you'd like to open a separate issue / PR for adding SOCKS support to core, we can continue that discussion there. :)

**Comment by 1dal** [comment]
Hi elazar, 
sorry, but im not sure how to write test for it correctly. :(

**Comment by 1dal** [comment]
@elazar, ok, I tried it.

**Comment by elazar** [comment]
Looks good to me. @matthewtrask @svpernova09 ?
